### PR TITLE
Add annotation konghq.com/host-aliases for adding additional hosts to Route Hosts

### PR DESCRIPTION
**Pull request by llinvillegwre**
**What this PR does / why we need it**
Adds a new annotation "konghq.com/host-aliases" for specifying additional hosts to append to Route Hosts. This allows one to add multiple Hosts to a Route without having to duplicate Routes across multiple Ingress Rules.

**Which issue this PR fixes**
Fixes issue #1164

**Special notes for your reviewer**
1. Adds new, optional annotation: konghq.com/host-aliases
2. New annotation will not allow duplicates
3. New annotation (when provided) appends to Route Hosts during annotation overrides

### CODE REVIEW

**Comment by rainest** [comment]
Minor issue with go.mod due to a manually-added module. You should remove `github.com/k0kubun/colorstring` from go.mod and tidy again.

Code generally looks fine implementation-wise, but I believe it proposes a practical challenge. Further discussion in #1164 since there's other history there.

**Review by shaneutt — APPROVED** [review]

File: pkg/annotations/annotations.go [hunk 1]
```diff
@@ -231,3 +232,12 @@ func ExtractResponseBuffering(anns map[string]string) (string, bool) {
...
 }
+
+// ExtractHostAliases extracts the host-aliases annotation value.
+func ExtractHostAliases(anns map[string]string) ([]string, bool) {
+	val, exists := anns[AnnotationPrefix+HostAliasesKey]
+	if val == "" {
+		return nil, exists
+	}
+	return strings.Split(val, ","), exists
+}
```
[the remark this answers is not in the document]
Comment (shaneutt) [reply pkg/annotations/annotations.go:243]: In the meantime if the end-user does manage to set the annotation to an empty string for some odd reason though, we can just return false here as it will be semantically equivalent to what we're doing now, but would be a functional improvement:

```suggestion
// ExtractHostAliases extracts the host-aliases annotation value.
func ExtractHostAliases(anns map[string]string) ([]string, bool) {
	val, exists := anns[AnnotationPrefix+HostAliasesKey]
	if !exists {
		return nil, false
	}
	if val == "" {
		return nil, false
	}
	return strings.Split(val, ","), true
}
```

File: pkg/annotations/annotations_test.go [hunk 2]
```diff
@@ -645,3 +645,36 @@ func TestExtractResponseBuffering(t *testing.T) {
...
+		},
+		{
+			name: "non-empty",
+			args: args{
+				anns: map[string]string{
+					"konghq.com/host-aliases": "foo.kong.com,bar.kong.com",
+				},
+			},
+			want: []string{"foo.kong.com", "bar.kong.com"},
+		},
```
Comment (shaneutt) [inline pkg/annotations/annotations_test.go:670]: Related to the above comment, we can add a small test case when the annotation is misconfigured as an empty string:

```suggestion
		},
		{
			name: "misconfigured",
			args: args{
				anns: map[string]string{
					"konghq.com/host-aliases": "",
				},
			},
			want: nil,
		},
```

**Comment by shaneutt** [comment]
The CI failure for license check is a misnomer, I checked manually :+1:

**Comment by rainest** [comment]
Thank you for your contribution. Please fill out the following form to claim your contributor swag:  
https://github.com/Kong/kong/blob/master/CONTRIBUTING.md#contributor-t-shirt.
